### ISSUE TITLE
Seeking Solution For Odoo/Gmail. Follow the leader e-mail distribution not cutting it.

### ISSUE THREAD

**Issue by dxfan227** [odoo/odoo#119773]
Odoo Online V14.

E-mail leaves ALOT to be desired within Odoo. The Follow the leader/e-mail routing is cumbersome and unclear for users on who is on an e-mail and who isn't, individuals cc'd by leads/customers are not displayed at all and require direct followup requesting their information to then be manually added as a follower, users must reply to a catchall instead of the message originator.

Are there any solutions that will allow odoo to act more as a traditional e-mail? As opposed to a redistribution/notification via e-mail with spoofing? Open to any/all ideas.

**Comment by XeJames** [comment]
Odoo is very blunt; Odoo is an email CONSUMER and is not meant to replace your traditional email.
Use Odoo mail only for "catchall" mail generated by Odoo.
To chat with clients, use your normal email client.

**Comment by dxfan227** [comment]
> Odoo is very blunt; Odoo is an email CONSUMER and is not meant to replace your traditional email.
> Use Odoo mail only for "catchall" mail generated by Odoo.
> To chat with clients, use your normal email client.

Yea, this is the standard Odoo support response, but is a cop out as all communication originates from Odoo. Whether a marketing outreach ( CRM ), a quotation ( Sales ), an invoice ( invoice) a purchase order ( Purchase). All of these start within Odoo no matter what. To expect the customer to switch communication and not reply, forward, or cc others to their original e-mail is asking a lot. On top of that, nothing would be logged within odoo if that was the case.

**Comment by XeJames** [comment]
#OpenSource

In this case, I don't agree with your assessment.  I went through this just like you and came to realize that the intention of the email connector in Odoo is only to allow you to track resource related "chatter" with people who do not have a portal account.  Nothing more.  You CAN open an email thread in Discuss but, again, this is not the purpose.

I looked into it a bit and there is CERTAINLY a need for an "Email App" which could piggyback on the existing platform, but nobody has written it yet.  But it's kind of rough to blame Odoo for the community not prioritizing a feature you wish it had, right?

1. I use my standard email client.  
2. I wish someone in the Odoo community would prioritize an email client.  
3. I am too lazy to do it myself - what's your excuse ;-)
4. 
#OpenSource

**Comment by dxfan227** [comment]
> #OpenSource
> 
> In this case, I don't agree with your assessment. I went through this just like you and came to realize that the intention of the email connector in Odoo is only to allow you to track resource related "chatter" with people who do not have a portal account. Nothing more. You CAN open an email thread in Discuss but, again, this is not the purpose.
> 
> I looked into it a bit and there is CERTAINLY a need for an "Email App" which could piggyback on the existing platform, but nobody has written it yet. But it's kind of rough to blame Odoo for the community not prioritizing a feature you wish it had, right?
> 
> 1. I use my standard email client.
> 2. I wish someone in the Odoo community would prioritize an email client.
> 3. I am too lazy to do it myself - what's your excuse ;-)
> 
> #OpenSource

If I was utilizing a 100 percent free system then I would agree completely, its on me and the community to make it happen. However this is not how corporate odoo is "selling" this program, we also pay a not insignificant amount on money every month to use Odoo + Odoo Studio+ "Success Packs" to add basic functionality that again was sold as easy and should work out of the box such as lot pricing on quotes. Out of the box, a customer can't even access the customer portal preview page until an internal users themselves clicks "Customer Preview" on the associated quote before sending. Other wise Token=False and they get a 404.  

Back to e-mail, current odoo reps are selling this as " just connect your e-mail and everything is tracked and logged for you so you can get back to it, we know how important record keeping is and this way you won't ever need to search through your e-mail for something" and " You can even just sign in with your google account ( this implementation is simply an alternative way to login to odoo via Oath and NOT a direct e-mail connect/feed as the reps are constantly saying it is). 


If I want to send an quotation to a customer.....you suggest I create the quote, export it as PDF, then send it directly in an e-mail ( where it won't be tracked by odoo)? Defeats the entire purpose of the system in the first place.  It's okay to admit that the chatter is a half baked attempt........it doesnt work as a notification system to "non portal" users either, as each non portal user would need to be manually added as a user record in odoo, then tagged to follow. On top of that, anyone else chiming in who is not already explicitatly tagged (CC'd by others)  as a follower will not be able to contribute to the conversation/receive updates.

**Comment by XeJames** [comment]
You may be misunderstanding.  Odoo DOES do exactly what you have described; the email connector ties DIRECTLY to the quotation itself and remains in the chatter along with the entire conversation.

All Odoo is not "great" at (email-wise) is when you initiate a random conversation with someone.  All Odoo email is "attached" to a resource of some type: Quotation, Invoice, Purchase Order... etc.

If you have an Odoo Partner to walk you through this, you might find more satisfying answers.  Odoo Enterprise is definitely not something you want to tackle without a good partner.

**Comment by dxfan227** [comment]
Do you have a recommendation for an odoo partner then? What you are describing has not been my experience of Odoo Online out of the box. 

The only "e-mails" it records are those it digests from catchall@ ( also very confusing for our customers and spam guards as odoo sets this a default reply to: which is flagged as different as originating e-mail + if the other parties DKIM/SPF forbids forwarding on behalf ( spoofing) as MOST domains do, odoo will fail the last mile delivery of the notification to other's inboxes). All of this is also only true if the messages originates with it's original e-mail headers + has no one CC'd + is originating from the creator of the record or a follower.

**Comment by XeJames** [comment]
I was quite worried about that initially as well.  But I run Odoo in a HA Cluster behind both a Firewall and a separate HAproxy and send/receive many emails daily through Odoo.   The biggest issue I face is being classified as junk mail - which is just a fact of life in business today.  So I'd say it's probably as reliable as any other email solution for business documents.

As for finding a good partner, that probably depends on where you are located; it's always best to have someone local - but not always possible.  You can to go https://www.odoo.com/partners and do a location based search.

If you cannot find someone suitable there, let me know and I'll try to hook you up with a suitable partner.

**Comment by diggy128** [comment]
I agree with the OP. Email handling in odoo is exactly as described. If all clients were technical and would stop replying, eg. To an invoice notification email to send a new order, it could work, at least for responses to odoo generated email.

As this is not a reality, in our company, we use a couple of older OCA modules fetchmail_attach_from_folder, mail_outbound_true_from that we ported internally to v15 and and a couple more we developed in-house.

All incoming emails are handled in the email clients, spam deleted, and email that need to be attached to an odoo record are moved to respective imap folders which odoo then parses, fetches and places in the chatter for the respective object.

Communicating with partners also happens from within odoo. Emails are sent from the chatter from the object in context (eg. communication regarding an order from that order's chatter).

This all has some clever heuristics that make the whole thing work for us, not perfectly, albeit 80% of what we needed without using the catchall.
